2.0/m_conn_accounts.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2018 Matt Schatz <genius3000@g3k.solutions>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* $ModAuthor: genius3000 */
+/* $ModAuthorMail: genius3000@g3k.solutions */
+/* $ModDesc: Limit SASL connect classes by account(s). */
+/* $ModDepends: core 2.0 */
+/* $ModConfig: Within connect block: accounts="sally22 BillBob someoneElse" */
+
+/* "accounts" is a space separated list of EXACT matching account names.
+ * This module requires that 'm_services_account' be loaded and will
+ * respect the "requireaccount" setting.
+ */
+
+#include "inspircd.h"
+#include "account.h"
+
+
+class ModuleConnAccounts : public Module
+{
+	const AccountExtItem* accountname;
+
+ public:
+	ModuleConnAccounts()
+	{
+		accountname = GetAccountExtItem();
```

Comment:
What happens to this if `m_services_account` unloads?

Reply:
Let me run another test or two on variations regarding this, but I did unload m_services_account and connect a client to match a class to this module and it denied it, as expected. I'll make sure it handles loading without okay too.

Reply:
Alright so no crashes any which way I tried but one scenario isn't great:
- starting Insp with this (or loading this while running) but not m_services_account: denies anyone to a class with `accounts`
   - loading m_services_account after: requires reload of this to get the ExtItem
- unloading m_services_account: same as starting without m_services_account
- reloading m_services_account: continues to work. 

I could go two routes to better this:
1. Remove the const and catch OnLoadModule to update the ExtItem.
2. Switch to getting the ExtItem each time within OnSetConnectClass, which is similar to how m_require_auth works.  

Easier and safer way would be #2 but increased number of ExtItem gets. Any suggested preference?